### DISCUSSION TITLE
ServerRpc not getting called..

### DISCUSSION THREAD

**Discussion by phoxman1982** [FirstGearGames/FishNet#223] [Q&A - DO NOT USE - ASK ON DISCORD] [answered]
I have this bit of code, I verified that all the arguments I pass into the serveAsset method call are all populated and valid.

>     public void loadAsset(string file, Vector3 pos, Quaternion rot, bool scene = false){
>          // signal to server that we uploaded an asset for spawning
>          Debug.Log("VIEWER - STAGE ONE");
>          serveAsset(file, EverVerse.Globals.rid, pos, rot, scene);
>     }

I then have this ServerRpc method which is called from the above code.

>     [ServerRpc(RequireOwnership = false)]
>     private void serveAsset(string file, string rid, Vector3 pos, Quaternion rot, bool scene = false, NetworkConnection sender = null){
>     Debug.Log("SERVER - STAGE ONE");
> 
>     // get the sender position and rotation, load the object
>     objData data = new objData();
>     data.scene = scene;
>     data.file = file;
>     data.rid = rid;
>     data.ext = file.Substring(file.LastIndexOf('.') + 1);
> 
>         if(!scene){
>             data.pos = sender.FirstObject.transform.position;
>             data.rot = sender.FirstObject.transform.rotation;
>         } else {
>             data.pos = pos;
>             data.rot = rot;
>         }
> 
>     Load(file, data, scene);
>     }
> 

The script is attached to the player prefab and is running for each player. The first player it works fine, the second player the ServerRpc never gets called. I figured it should work since I (RequireOwnership = false) in the ServerRpc call, but it doesn't, so why would that be?

I tried searching but due to the relative newness of Fish-Net, almost nothing comes up for issues when searching Google.
Thanks for any help in advance!

**Comment by FirstGearGames** [comment]
Have you tried asking on our Discord server yet? I really shouldn't have a QA on the git because they get overlooked so fast.

**Reply by phoxman1982** [reply]
Yes I have but not a single person responded yet, I even made it a thread nice and pretty.
It's in Wild West channel. Thanks :)

**Comment by FirstGearGames** [comment]
Gotcha. Probably was missed. Feel free to mention me in your thread. I'm
off for tonight and tomorrow is my day off as well, so if I don't get back
to you Tuesday feel free to ping me again.

**Comment by phoxman1982** [comment, marked as answer]
So apparently this was fixed when I upgraded to 3.1.5, so all good now, cheers!

Sent with [Proton Mail](https://proton.me/) secure email.

------- Original Message -------
